llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -8412,8 +8412,35 @@ void ScalarEvolution::visitAndClearUsers(
     SmallVectorImpl<const SCEV *> &ToForget) {
   while (!Worklist.empty()) {
     Instruction *I = Worklist.pop_back_val();
-    if (!isSCEVable(I->getType()))
+    if (!isSCEVable(I->getType())) {
```

Comment:
```suggestion
    if (!isSCEVable(I->getType()) && !isa<WithOverflowInst>(I)) {
```
should be sufficient, no?